client/extensions/woocommerce/components/form-click-to-edit-input/index.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormTextInput from 'components/forms/form-text-input';
+
+class FormClickToEditInput extends Component {
+
+	static propTypes = {
+		onChange: PropTypes.func,
+		value: PropTypes.string,
+		placeholder: PropTypes.string,
+		disabled: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		onChange: noop,
+		placeholder: '',
+		value: '',
+		disabled: false,
+	};
+
+	state = {
+		isEditing: false,
+		value: '',
+	};
+
+	editStart = () => {
+		const { value } = this.props;
+		this.setState( {
+			isEditing: true,
+			value
+		} );
+	}
+
+	editEnd = () => {
+		const { onChange } = this.props;
+		if ( this.props.value !== this.state.value ) {
+			onChange( this.state.value );
+		}
+
+		this.setState( {
+			isEditing: false,
+			value: '',
+		} );
+	}
+
+	onInputChange = ( e ) => {
+		this.setState( {
+			value: e.target.value
+		} );
+	}
+
+	renderInput() {
+		const props = { ...this.props,
+			onChange: this.onInputChange,
+			value: this.state.value,
+		};
+		return (
+			<span className="form-click-to-edit-input__wrapper editing">
+				<FormTextInput
+					{ ...props }
+					autoFocus
+					className="form-click-to-edit-input__input"
+				/>
+				<Button
```

Comment:
This `Button` could use `aria-label={ translate( 'Update SKU' ) }`, to help future accessibility efforts.